### ISSUE TITLE
Feature request: RSA Encryption

### ISSUE THREAD

**Issue by jonathansm1th** [hairyhenderson/gomplate#800]
We need to add `rsa` to the crypto functions to enable our git-ops approach to Terraform secrets.

TLDR;
We're using gomplate to render templated terraform files and are moving towards a more git-ops approach of encrypting secrets inline instead of storing them one-at-a-time in [Vault](https://www.vaultproject.io). This enables us to generate `.tf` files which do not require any additional tooling at execution, using Terraform's only decryption method: [rsadecrpt](https://www.terraform.io/docs/configuration/functions/rsadecrypt.html) 

_Our work-around_ is to use `openssl rsautl -encrypt` to encrypt new terraform readable secrets, (which are stored as part of a `config.yaml` datasource); which worked fine, until we needed to encrypt a bunch of secrets extracted from an api's json response.

Before we jump on a PR, what are your thoughts?

**Comment by hairyhenderson** [comment]
Thanks @jonathansm1th - great suggestion!

How about a function in the `crypto` namespace, like `crypto.RSAEncrypt`?

This might be as simple as using the original `rsaencrypt` code from the Terraform PR where `rsadecrypt` was added:

https://github.com/hashicorp/terraform/commit/9263e335c4a084de49b4f500b6f3225629833654#diff-38c6c95b4ed4e10a18cdf045d73d94dbR1714-R1733

For completeness, it would probably also be good to add a `crypto.RSADecrypt` function.

Contrary to the function signature in Terraform, to make pipelining useful these functions should take the key (private key for encrypt, public key for decrypt) as the _first_ parameter.

So usage would be something like:

```go
{{- $privKey := .secrets.privKey }}
{{- $pubKey := .secrets.pubKey }}
{{- $cyphertext := crypto.RSAEncrypt $privKey "super secret text" }}
{{- $cyphertext | crypto.RSADecrypt $pubKey }}
```

And I'd expect this to work thus:

```console
$ gomplate -c secrets=secrets.json -f input.tmpl
super secet text
```

And as far as formatting, I'd expect that the private and public keys would be in PEM format (base64-encoded).

And for the input/output strings, treat them as just plain byte arrays  - if `crypto.RSADecrypt` produces binary output, it'd be the template author's responsibility to pipe that through `base64.Encode` if necessary.
